src/main.rs

```diff
@@ -456,3 +468,22 @@ fn flashing_progress() -> flashing::FlashProgress {
         }
     })
 }
+
+fn get_stack_start_and_reset_handler(
+    sess: &mut Session,
+    elf: &Elf,
+) -> anyhow::Result<(u32, Range<u32>)> {
+    let mut core = sess.core(0)?;
+    core.reset_and_halt(Duration::from_secs(5))?;
```

Comment:
is it necessary to do a reset halt *here*? there's another call to reset_and_halt in `main.rs` due to `Canary::install` so this may result in the device being reset twice, which should not be a problem to the operation of probe-run but it's preferable to not do double work.

perhaps the reset_and_halt can be do once before calling this function, then it can also be removed from canary::install

Reply:
I will address this in the follow-up clean-up PR, since I will change the order of operations there and also touch the canary